### ISSUE TITLE
[Doc]: Better strategy to remove PBC condition for Ligands in the complex

### ISSUE THREAD

**Issue by Dead-fisher** [Valdes-Tresanco-MS/gmx_MMPBSA#353]
### Documentation Link

https://valdes-tresanco-ms.github.io/gmx_MMPBSA/dev/gmx_MMPBSA_running/

### Problem

Hi!

The doc I mention states that the input of the `gmxMMPBSA` requires the removal of the PBC condition.

The doc also gives the typical command line as:
```
# remove PBC
gmx trjconv -s md.tpr -f md.xtc -o md_noPBC.xtc -pbc mol -center -n -ur compact
center: 20 (created group)
output: 0
# fit
gmx trjconv -s md.tpr -f md_noPBC.xtc -o md_fit.xtc -n -fit rot+trans
fit: 20 (created group)
output: 0
```

However, this command still can't align all the trajectories to the reference, leading to the ligands' RMSD jump.
(My `gromacs` version is `2022.3`)

I am not very sure how this will effect the result of `gmxMMPBSA`.

So, as for the `PBC` condition in the doc, does only the "broken" molecule affect the calculation? or the "broken" complex (where the protein and ligand are in separate cells) can also affect the calculation?

### Suggested improvement

With the suggestion from the `gromacs` doc, I guess:
```
gmx trjconv -s ref.tpr -f md.xtc -pbc nojump -center -o nopbc.xtc -n index.ndx
```
Basically, the `-pbc nojump` can check all the PBC crossing issues. which gives the visually continuous trajectory.

**Comment by Valdes-Tresanco-MS** [comment]
Commonly, `-pbc mol` solves the problem if the box is defined large enough so that in any conformation, the molecule doesn't get out. However, using `-pbc nojumps` may be a more appropriate choice since we do not know how the box is configured. We will follow your recommendations

**Comment by marioernestovaldes** [comment]
There is nothing written in stone when it comes to removing PBC conditions.  As previously stated, `-pbc mol` should suffice in most cases, and is "less harsh" than "-pbc nojump". Nevertheless, the recommendation is valid :)

**Comment by marioernestovaldes** [comment]
updated docs https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/commit/7574ff6ecf2e8ed76aec647c52005a9d53c4bed6

closing issue...

**Comment by Dead-fisher** [comment]
Thanks, yeah, dealing with PBC is kinda tricky.
When we handle only a few systems with `gmx_MMPBSA`, we can visually check the quality of the trajectories. But when handling a bunch of systems, which is my case, a few frames might have RMSD jump previously. I just found this by checking the RMSD plot of the ligands. 
I guess a piece of warning info from the RMSD monitor or just mandatorily removing the PBC could be a better way for the batch works.
Anyway, thanks for accepting me.

**Comment by hackerzone85** [comment]
Hi, 
Yanze Wang
Handling periodic boundary conditions (PBC) correctly is crucial for accurate molecular dynamics simulations and visualizations. The -pbc option in the gmx trjconv command specifies how to handle periodic boundary conditions (PBC). Here are the meanings of each choice:
*none*: No PBC corrections are applied. The trajectory remains as it is.
*mol*: Ensures that molecules are whole by removing any PBC artifacts. This is useful for visualizing molecules without parts appearing on opposite sides of the box.
*res*: Similar to mol, but ensures that residues are whole. This is useful for systems where residues might span across periodic boundaries.
*atom*: Ensures that atoms are within the primary unit cell. This option is rarely used.
*nojump*: Corrects for molecules that have jumped across the periodic boundaries, ensuring a smooth trajectory without sudden jumps.
*cluster*: Groups molecules that are within a certain distance of each other, effectively clustering them together.
whole: Ensures that molecules are whole and places them in the primary unit cell.

As per your suggestion, using the combination "-pbc mol" ensures that the entire protein and ligand are whole and "-pbc nojump" corrects for any jumps across periodic boundaries, providing a smooth and continuous trajectory.

However, I still have doubts regarding group selection in the case of protein-ligand complex. The document you shared suggested group selection for centering is "20 (Protein-Ligand Group; Created group)". The J.A. Lemkul (http://www.mdtutorials.com/gmx/complex/09_analysis.html) suggested group selection for centering is "Protein" and "System" for output. He also mentioned  that centering complexes (protein-ligand, protein-protein) may be difficult for longer simulations involving many jumps across periodic boundaries. In those instances (particularly in protein-protein complexes), it may be necessary to create a custom index group to use for centering, corresponding to the active site of one protein or the interfacial residues of one monomer in a complex.

I need your expert opinion on group selection.

***************************************************************************************************************

Here is command I was using for my study
Protein-Ligand Complex: Cleaning trajectory and extracting the first frame:
1. Remove periodic boundary conditions (PBC) to ensure that molecules are whole and water molecules from the trajectory:
```
gmx trjconv -s md_0_100.tpr -f md_0_100.xtc -o md_0_100_noPBC_noWAT.xtc -pbc mol -ur compact -center -n index.ndx
```
Select group for centering: *1 (Protein)*
Select group for output: *19 (Protein_Ligand)*

2. Remove periodic boundary conditions (PBC) to correct for any jumps across periodic boundaries to provide a smooth trajectory and water molecules from the trajectory:
```
gmx trjconv -s md_0_100.tpr -f md_0_100_noPBC_noWAT.xtc -o md_0_100_noPBC_noWAT_noJUMP.xtc -pbc nojump -ur compact -center -n index.ndx
```
Select group for centering: *1 (Protein)* 
Select group for output: *19 (Protein_Ligand)*

3. Performs a least squares fit to a reference structure (C-alpha atoms) and centers the trajectory:
```
gmx trjconv -s md_0_100.tpr -f md_0_100_noPBC_noWAT_noJUMP.xtc -o md_0_100_noPBC_noWAT_noJUMP_fit.xtc -ur compact -center -fit rot+trans -n index.ndx
```
Select group 3 (C-alpha) for fitting.
Select group for centering:  *1 (Protein)*
Select group for output: *19 (Protein_Ligand)*

4. Extracts the first frame as a PDB file:
```
gmx trjconv -s md_0_100.tpr -f md_0_100_noPBC_noWAT_noJUMP_fit.xtc -o first_frame.pdb -dump 0 -n index.ndx
```
Select group  for output: *19 (Protein_Ligand)*

**Comment by Valdes-Tresanco-MS** [comment]
@hackerzone85 Your procedure is completely correct. As for the groups to select for fitting, the protein is generally chosen in protein-ligand type complexes simulation. This is because it is larger and therefore, it is expected that if it is fine, then the ligand should also be fine. Also because it is possible that the ligand is unbound and is not close to the initial position, therefore it would be very difficult to fit it to its initial structure. However, if you see that there are no considerable variations in sight, it is also valid to use the complex.
As for the procedure, note you are removing the water molecules in step one by selecting only the complex as output. In case you wanted to carry out an analysis that involves them, they would not be present.
In case you have questions please open a discussion and cite this issue, since it has been closed for a long time it is difficult to follow up on it.
